src/components/orator/js/Orator.js

```diff
@@ -0,0 +1,571 @@
+        selectors: {
+            controller: ".flc-orator-controller",
+            content: ".flc-orator-content"
+        },
+        model: {
+            play: false
+        },
+        components: {
+            controller: {
+                type: "fluid.orator.controller",
+                options: {
+                    container: "{orator}.dom.controller",
+                    scope: "{orator}.container",
+                    model: {
+                        playing: "{orator}.model.play"
+                    }
+                }
+            },
+            domReader: {
+                type: "fluid.orator.domReader",
+                container: "{that}.dom.content",
+                options: {
+                    listeners: {
+                        "{tts}.events.utteranceOnEnd": [{
+                            changePath: "{orator}.model.play",
+                            value: false,
+                            source: "domReader.tts.utteranceOnEnd",
+                            priority: "after:removeHighlight",
+                            namespace: "domReader.stop"
+                        }]
+                    },
+                    modelListeners: {
+                        "{orator}.model.play": {
+                            funcName: "fluid.orator.handlePlayToggle",
+                            args: ["{that}", "{change}.value"],
+                            namespace: "domReader.handlePlayToggle"
+                        }
+                    }
+                }
+            }
+        },
+        distributeOptions: [{
+            source: "{that}.options.controller",
+            target: "{that controller}.options",
+            removeSource: true,
+            namespace: "controllerOpts"
+        }, {
+            source: "{that}.options.domReader",
+            target: "{that domReader}.options",
+            removeSource: true,
+            namespace: "domReaderOpts"
+        }]
+    });
+
+    fluid.orator.handlePlayToggle = function (that, state) {
+        if (state) {
+            that.play();
+        } else {
+            that.pause();
+        }
+    };
+
+    /**********************************************
+     * fluid.orator.controller
+     *
+     * Provides a UI Widget to control the Orator
+     **********************************************/
+
+    fluid.defaults("fluid.orator.controller", {
+        gradeNames: ["fluid.newViewComponent"],
+        selectors: {
+            playToggle: ".flc-orator-controller-playToggle"
+        },
+        styles: {
+            play: "fl-orator-controller-play"
+        },
+        strings: {
+            play: "play",
+            pause: "pause"
+        },
+        container: "",
+        scope: "",
+        model: {
+            playing: false
+        },
+        members: {
+            container: "@expand:fluid.orator.controller.container({that}.options.container, {that}.options.markup.defaultContainer, {that}.options.scope)"
```

Comment:
This is somewhat abuse of the expected workflow of "newViewContainer" which is effectively the standard one in the upcoming framework. The linkage between options.container and members.container should be standard in the base grade, and all actions which are necessary to establish it pushed into the evaluation of options.container, for example like this: https://github.com/fluid-project/fluid-authoring/blob/FLUID-4884/src/js/ComponentGraph.js#L581

Reply:
@amb26 and I discussed this in more detail today. The definition of the "container" should be in the base grade and not overridden. Rather we can have the expander refer to an invoker where any necessary changes can be applied.

Reply:
Align the structure more with what's in "fluid.author.containerRenderingView", see link above. And add an additional invoker for rendering into the DOM.. e.g to allow for prepending or appending into the parentContainer.